tests/foreman/cli/test_realm.py

```diff
@@ -29,128 +29,118 @@
 from robottelo.decorators import run_in_one_thread
 from robottelo.decorators import tier1
 from robottelo.decorators import tier2
-from robottelo.test import CLITestCase
+
+
+@pytest.fixture(scope='module')
+def _make_proxy(options=None):
+    """Create a Proxy and register the cleanup function"""
+    proxy = make_proxy(options=options)
+    yield proxy
+    capsule_cleanup(proxy['id'])
 
 
 @run_in_one_thread
-class RealmTestCase(CLITestCase):
+class TestRealm:
     """Tests for Realms via Hammer CLI, must be run on QE Satellite Server.
     Requires enroll_idm() and configure_realm() to configure the test environment."""
 
-    def _make_proxy(self, options=None):
-        """Create a Proxy and register the cleanup function"""
-        proxy = make_proxy(options=options)
-        # Add capsule to cleanup list
-        self.addCleanup(capsule_cleanup, proxy['id'])
-        return proxy
-
-    def setUp(self):
-        self.realm = None
-        self.realm_name = gen_string('alpha', random.randint(1, 30))
-
-    def tearDown(self):
-        """Delete realm from proxy after test completes"""
-        if self.realm:
-            self.addCleanup(realm_cleanup, self.realm['id'])
-
     @tier1
-    def test_positive_delete_by_name(self):
+    def test_positive_delete_by_name(self, _make_proxy):
         """Realm deletion by realm name
 
         :id: ef3967e6-d53d-4dec-b74f-c20448d5fc6d
 
         :expectedresults: Realm is deleted
         """
-        proxy = self._make_proxy()
         realm = make_realm({'realm-proxy-id': proxy['id'], 'realm-type': 'Active Directory'})
```

Comment:
```suggestion
        realm = make_realm({'realm-proxy-id': _make_proxy['id'], 'realm-type': 'Active Directory'})
```
?
flake8 on travis complains there's undefined variable `proxy` here.